### DISCUSSION TITLE
Create a playlist

### DISCUSSION THREAD

**Discussion by insomniachi** [SuRGeoNix/Flyleaf#534] [Q&A]
I have a list of URL's how do i create a playlist from that ?
this is what i tried
```
var playlist = MediaPlayer.Playlist;

foreach (var item in response.Items.Where(x => x.UserData?.Played is null or false))
{
	var uri = await jellyfinClient.GetMediaUrl(item);
	if(uri is null)
	{
		continue;
	}

	playlist.AddItem(new PlaylistItem
	{
		DirectUrl = uri.ToString(),
		Episode = item.IndexNumber ?? -1,
		Enabled = true,
		Title = item.Name
	}, "");
}

if(playlist.Items.Count == 0)
{
	return false;
}

var args = MediaPlayer.Open(playlist.Items.First());
```

but i get this exception
![image](https://github.com/user-attachments/assets/ddccab17-f273-4920-a74d-620bbfee42ec)

**Comment by SuRGeoNix** [comment]
Hi @insomniachi,

This playlist is only for plugins (it shouldn't be public but as the plugins are external assemblies it has to be).

Unfortunately Flyleaf currently does not support Playlists in the way that you want to. You could either create your own Playlist handler or possible a plugin.

I might try to add this to the default plugin in the next update to support Playlists though.
